src/util/getCommandState.ts

```diff
@@ -16,6 +16,27 @@ const tags = {
   italic: createTag('i'),
   underline: createTag('u'),
   strikethrough: createTag('strike'),
+  foreColor: createTag('font'),
+  backColor: createTag('span'),
+}
+
+/** Extracts the foreground and background colors from the given string.
+ * Returns an object with foreColor and backColor properties.
+ * If the string does not contain a font or span tag, the default colors are returned.
+ */
+const extractColors = (savedValue: string) => {
+  const foreColorRegex = /<font[^>]*\scolor=["']?([^"']+)["']?[^>]*>/i
+  const backColorRegex = /<span[^>]*\sstyle=["'][^"']*background-color:\s*([^;"']+)/i
+
+  // Attempt to extract the font color
+  const foreColorMatch = savedValue.match(foreColorRegex)
+  const foreColor = foreColorMatch ? foreColorMatch[1].trim() : 'rgb(227, 227, 227)'
```

Comment:
The specific foreground/background color we use will depend on the current theme, so we can't hardcode these here unfortunately.